address_space_integrity/address_space_integrity.md

```diff
@@ -0,0 +1,588 @@
+<!-- Output copied to clipboard! -->
+
+
+# Address Space Protection for FuSa
+
+
+## Kernel and architecture protection
+
+Christoffer Hall-Frederiksen
+
+Gabriele Paoloni
+
+Alessandro Carminati
+
+Eric Chanudet
+
+With help from the Red Hat MM team.
+
+
+
+
+# 
+
+
+# Contents
+
+
+[TOC]
+
+
+
+
+
+# Abbreviations
+
+
+<table>
+   <td>FuSa
+   </td>
+   <td>Functional Safety
+   </td>
+  </tr>
+  <tr>
+   <td>MMU
+   </td>
+   <td>Memory Management Unit
+   </td>
+  </tr>
+  <tr>
+   <td>TLB
+   </td>
+   <td>Translation Lookaside Buffer
+   </td>
+  </tr>
+  <tr>
+   <td>VMA
+   </td>
+   <td>Virtual Memory Area
+   </td>
+  </tr>
+</table>
+
+
+
+# 
+
+
+# Introduction
+
+This document derives from the [investigation](https://docs.google.com/document/d/1JrjBHuYopU9Z7xOwm7KOc5DTKzRnvjk-NIY2QfBlBs0/edit?userstoinvite=qclzdh%40gmail.com&actionButton=1) that started in the ELISA Safety Architecture WG. The high level safety goal is  \
+&lt;<To maintain and enforce the integrity of the process address space along with the process lifecycle or detect an integrity violation failure (system level measures or additional mechanisms in the Kernel)>>
+
+
+# Safety Goals
+
+**SG01:** The Kernel shall properly create the address space of the safety critical application when the associated process is spawned.
+
+**SG02: **The Kernel shall  maintain the integrity of the safety application process descriptor, process address spaces and associated structures at runtime when memory is allocated, used and freed.
```

Comment:
From ww19 Safety Arch WG meeting it came out that we should also consider the settings of cred struct and its impact on the safety goal

ref: https://elixir.bootlin.com/linux/latest/source/include/linux/cred.h#L110